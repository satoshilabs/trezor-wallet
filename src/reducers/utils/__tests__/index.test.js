import * as reducerUtils from '../index';

describe('reducers utils', () => {
    it('observeChanges shoud be the same - returns false', () => {
        const data = [
            // example of same data (false)
            {
                pervious: {},
                current: {},
            },
            {
                pervious: 1,
                current: 1,
            },
            {
                pervious: [],
                current: [],
            },
            {
                pervious: [1, 1, 1],
                current: [1, 1, 1],
            },
            {
                pervious: 'a',
                current: 'a',
            },
            {
                pervious: { one: 1 },
                current: { one: 1 },
            },
            {
                pervious: { one: { two: 1 } },
                current: { one: { two: 1 } },
            },
            {
                pervious: { one: { two: [1, 2, 3] } },
                current: { one: { two: [1, 2, 3] } },
            },
            {
                pervious: { one: { two: [1, { three: 1 }, 3] } },
                current: { one: { two: [1, { three: 1 }, 3] } },
            },
            {
                pervious: { one: { two: [1, { three: 1 }, { four: 3, five: { six: 3 } }] } },
                current: { one: { two: [1, { three: 1 }, { four: 3, five: { six: 3 } }] } },
            },
        ];

        data.forEach((item) => {
            expect(reducerUtils.observeChanges(
                item.pervious, item.current,
            )).toMatchSnapshot();
        });
    });

    it('observeChanges shoul NOT be the same - returns true', () => {
        const data = [
            // example of different data (true)
            {
                pervious: { one: 1 },
                current: {},
            },
            {
                pervious: { one: 1 },
                current: { one: 1, two: 2 },
            },
            {
                pervious: [{}, {}],
                current: [],
            },
            {
                pervious: [1, 1, 1],
                current: [1, 1],
            },
            {
                pervious: 'a',
                current: 'b',
            },
            {
                pervious: 1,
                current: '1',
            },
            {
                pervious: { one: 1 },
                current: { one: 2 },
            },
            {
                pervious: { one: { two: 1 } },
                current: { one: { two: 2 } },
            },
            {
                pervious: { one: { two: 1 } },
                current: { one: { two: 2 } },
            },
            {
                pervious: { one: { two: [1, 2, 3] } },
                current: { one: { two: [1, 1, 3] } },
            },
            {
                pervious: { one: { two: [1, { three: 1 }, 3] } },
                current: { one: { two: [1, { three: 2 }, 3] } },
            },
            {
                pervious: { one: { two: [1, { three: 1 }, { four: 3, five: { six: 3 } }] } },
                current: { one: { two: [1, { three: 1 }, { four: 3, five: { six: 1 } }] } },
            },
            {
                pervious: { one: { two: [1, { three: 1 }, { four: 3, five: { sixxx: 3 } }] } },
                current: { one: { two: [1, { three: 1 }, { four: 3, five: { six: 1 } }] } },
            },
        ];

        data.forEach((item) => {
            expect(reducerUtils.observeChanges(
                item.pervious, item.current,
            )).toMatchSnapshot();
        });
    });
});
