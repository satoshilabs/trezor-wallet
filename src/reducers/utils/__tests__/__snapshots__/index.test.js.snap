// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers utils observeChanges shoud be the same - returns false 1`] = `false`;

exports[`reducers utils observeChanges shoud be the same - returns false 2`] = `false`;

exports[`reducers utils observeChanges shoud be the same - returns false 3`] = `false`;

exports[`reducers utils observeChanges shoud be the same - returns false 4`] = `false`;

exports[`reducers utils observeChanges shoud be the same - returns false 5`] = `false`;

exports[`reducers utils observeChanges shoud be the same - returns false 6`] = `false`;

exports[`reducers utils observeChanges shoud be the same - returns false 7`] = `false`;

exports[`reducers utils observeChanges shoud be the same - returns false 8`] = `false`;

exports[`reducers utils observeChanges shoud be the same - returns false 9`] = `false`;

exports[`reducers utils observeChanges shoud be the same - returns false 10`] = `false`;

exports[`reducers utils observeChanges shoul NOT be the same - returns true 1`] = `true`;

exports[`reducers utils observeChanges shoul NOT be the same - returns true 2`] = `true`;

exports[`reducers utils observeChanges shoul NOT be the same - returns true 3`] = `true`;

exports[`reducers utils observeChanges shoul NOT be the same - returns true 4`] = `true`;

exports[`reducers utils observeChanges shoul NOT be the same - returns true 5`] = `true`;

exports[`reducers utils observeChanges shoul NOT be the same - returns true 6`] = `true`;

exports[`reducers utils observeChanges shoul NOT be the same - returns true 7`] = `true`;

exports[`reducers utils observeChanges shoul NOT be the same - returns true 8`] = `true`;

exports[`reducers utils observeChanges shoul NOT be the same - returns true 9`] = `true`;

exports[`reducers utils observeChanges shoul NOT be the same - returns true 10`] = `true`;

exports[`reducers utils observeChanges shoul NOT be the same - returns true 11`] = `true`;

exports[`reducers utils observeChanges shoul NOT be the same - returns true 12`] = `true`;

exports[`reducers utils observeChanges shoul NOT be the same - returns true 13`] = `true`;

exports[`reducers utils observeChanges shoul NOT be the same - returns true 14`] = `true`;

exports[`reducers utils observeChanges shoul NOT be the same - returns true 15`] = `true`;

exports[`reducers utils observeChanges test filter 1`] = `true`;

exports[`reducers utils observeChanges test filter 2`] = `false`;
